src/psyclone/psyGen.py

```diff
@@ -4490,6 +4506,252 @@ def nodes_to_code_block(parent, statements):
         del statements[:]
         return code_block
 
+    def generate_schedule(self, name, module_ast):
+        '''
+        Create a KernelSchedule from the supplied fparser2 AST.
+
+        :param str name: Name of the subroutine representing the kernel.
+        :param module_ast: fparser2 AST of the full module where the kernel \
+                           code is located.
+        :type module_ast: :py:class:`fparser.two.Fortran2003.Program`
+        :raises GenerationError: Unable to generate a kernel schedule from the
+                                 provided fpaser2 parse tree.
+        '''
+        from fparser.two import Fortran2003
+
+        def first_type_match(nodelist, typekind):
+            '''
+            Returns the first instance of the specified type in the given
+            node list.
+
+            :param list nodelist: List of fparser2 nodes.
+            :param type typekind: The fparse2 Type we are searching for.
+            '''
+            for x in nodelist:
+                if isinstance(x, typekind):
+                    return x
+            raise ValueError  # Type not found
+
+        def search_subroutine(nodelist, searchname):
+            '''
+            Returns the first instance of the specified subroutine in the given
+            node list.
+
+            :param list nodelist: List of fparser2 nodes.
+            :param str searchname: Name of the subroutine we are searching for.
+            '''
+            for x in nodelist:
+                if (isinstance(x, Fortran2003.Subroutine_Subprogram) and
+                   str(x.content[0].get_name()) == searchname):
+                    return x
+            raise ValueError  # Subroutine not found
+
+        new_schedule = KernelSchedule(name)
+
+        try:
+            # Assume just 1 Fortran module definition in the file
+            if len(module_ast.content) > 1:
+                raise GenerationError("Unexpected AST when generating '{0}' "
+                                      "kernel schedule. Just one "
+                                      "module definition per file supported."
+                                      "".format(name))
+            mod_content = module_ast.content[0].content
+            mod_spec = first_type_match(mod_content,
```

Comment:
pylint complains that `mod_spec` is unused. Perhaps just do `_ = first_type_match(...)`

Reply:
I am not using this at the moment, it is to access the metadata, I commented it and added a TODO referencing #288